### PR TITLE
Use twig to render forms

### LINKED ISSUE #4050: Use twig blocks for form rendering
We should replace all `<form>` elements in the twig templates with the twig function `form_start` and `form_end`. We should also move the `action` and `method` (POST, GET) attribute to the PHP files.

This would decouples the business logic from the layout.
## Todo

The following files contains form elements.
- [ ] https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Resources/views/standard_layout.html.twig
- [ ] https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Resources/views/CRUD/base_acl_macro.html.twig
- [ ] https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Resources/views/CRUD/base_edit_form.html.twig
- [ ] https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Resources/views/CRUD/base_list.html.twig
- [ ] https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Resources/views/CRUD/batch_confirmation.html.twig
- [ ] https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Resources/views/CRUD/delete.html.twig

### PR DESCRIPTION
**Pull request by core23**
## Subject

Use twig to render all forms, this way we can respect the form themes.

I am targeting this branch, because this is BC.

Closes #4050, #4069

## Changelog

```markdown
### Changed
- Use twig to render forms
- Do not render hidden labels on confirmation page
```

### CODE REVIEW

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_edit_form.html.twig:12]: This is missing

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_edit_form.html.twig:15]: this is missing

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_edit_form.html.twig:17]: this is missing

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_edit_form.html.twig:18]: we're losing this block

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_edit_form.html.twig:14]: we're losing the block

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_acl_macro.html.twig:15]: this is missing

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_acl_macro.html.twig:17]: this is missing

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_acl_macro.html.twig:13]: the class is missing

File: src/Resources/views/standard_layout.html.twig
Comment (VincentLanglet) [inline src/Resources/views/standard_layout.html.twig]: Can you be more explicit in the comment ?

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (phansys) [inline src/Resources/views/CRUD/base_acl_macro.html.twig:64]: Should not we also try to replace this input?

File: src/Resources/views/CRUD/base_list.html.twig
Comment (phansys) [inline src/Resources/views/CRUD/base_list.html.twig:44]: Should not we also try to replace this input?

File: src/Resources/views/CRUD/base_list.html.twig
Comment (phansys) [inline src/Resources/views/CRUD/base_list.html.twig:384]: Should not we also try to replace this input?

File: src/Resources/views/CRUD/batch_confirmation.html.twig
Comment (phansys) [inline src/Resources/views/CRUD/batch_confirmation.html.twig:52]: Should not we also try to replace these inputs?

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_edit_form.html.twig:15]: This should automatically be rendered by the `form_start` function

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_acl_macro.html.twig:13]: This is a BC break / fix, depending on your application.

It *fixes* the ignored form theme of the `form` and now respects the theme you have set for your admin.

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_edit_form.html.twig:14]: Thats right, but overriding the form action with a block looks weird. You should override the action when creating the form inside the controller.

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_acl_macro.html.twig:15]: This is not needed, because symfony forms will add this

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_acl_macro.html.twig:17]: Moved this to the `form_attr` variable

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_acl_macro.html.twig:64]: Yes, but this could require more refactoring, e.g. we must check if the PHP form contains a button or not. Otherwise we could render the button twice or never.

File: src/Resources/views/CRUD/base_list.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_list.html.twig:44]: Same here: https://github.com/sonata-project/SonataAdminBundle/pull/7727/files#r822988289

File: src/Resources/views/CRUD/base_list.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_list.html.twig:384]: Same here: https://github.com/sonata-project/SonataAdminBundle/pull/7727/files#r822988289

File: src/Resources/views/CRUD/batch_confirmation.html.twig
Comment (core23) [reply src/Resources/views/CRUD/batch_confirmation.html.twig:52]: Same here: https://github.com/sonata-project/SonataAdminBundle/pull/7727/files#r822988289

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_edit_form.html.twig:12]: This is some kind of bugfix, because we now respect the form theme of the admin

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_edit_form.html.twig:17]: Moved this to the `form_attr` variable

File: src/Resources/views/CRUD/base_edit_form.html.twig
Comment (core23) [reply src/Resources/views/CRUD/base_edit_form.html.twig:18]: We cannot merge the content of the block with the `form_attr`. 

Maybe we can risk this special kind of BC?

File: src/Resources/views/CRUD/base_acl_macro.html.twig
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_acl_macro.html.twig:14]: What is needed to move it already to the controller ?

File: src/Resources/views/standard_layout.html.twig
Comment (VincentLanglet) [inline src/Resources/views/standard_layout.html.twig:201]: What's blocking to do it now ?

Comment (core23) [reply src/Resources/views/standard_layout.html.twig:201]: It requires a little bit more refactoring, because this template is loaded for every admin page.

We should either inject the new form using something like this https://github.com/sonata-project/SonataAdminBundle/pull/7482/files or we could render a sonata block.

Comment (VincentLanglet) [reply src/Resources/views/standard_layout.html.twig:201]: For having resolved lot of NEXT_MAJOR during the release of SonataAdmin 4, I don't like the idea of adding a NEXT_MAJOR comment that complicated. We generally try to add comment like "Remove this line" or "Uncomment this code", because it can be really hard for a dev to resolve todo, months later and without context/strategy to solve them.

Same for the `{# NEXT_MAJOR: Move logic to the controller #}` comment.

Can't we provide a simpler migration path for the futur sonata maintainer ? If not, it shouldn't be a next major comment but an issue instead.

**Comment by VincentLanglet** [comment]
Do you have time to finish this PR @core23 ?

**Pull request closed without merging**
